coreos-ostree-importer/README.md

```diff
@@ -0,0 +1,115 @@
+to point to your local server. For example: `amqp_url = "amqp://192.168.121.2"`
+See the [later section](#running-rabbitmq-server-locally)
+on that topic and update your fedora messaging config accordingly.
+
+From your local git directory:
+
+```
+podman build -t coreos-ostree-importer .
+```
+
+Create a file with aws credentials somewhere:
+
+```                                 
+cat <<'EOF' > /dev/shm/secret
+[default]
+aws_access_key_id=keyid
+aws_secret_access_key=key
+EOF
+```
+
+Create some empty OSTree repos:
+
+```
+mkdir /srv/prodrepo
+mkdir /srv/composerepo
+
+ostree --repo=/srv/prodrepo init --mode=archive 
+ostree --repo=/srv/composerepo init --mode=archive
+```
+
+Run the importer:
+
+```
+podman run -it --rm                                                \
+           -v $PWD/:/pwd/                                          \
+           -v /dev/shm/secret:/.aws/config                         \
+           -e AWS_CONFIG_FILE=/.aws/config                         \
+           -v /srv/composerepo/:/mnt/koji/compose/ostree/repo/:z   \
+           -v /srv/prodrepo/:/mnt/koji/ostree/repo/:z              \
+           coreos-ostree-importer
+```
+
+
+If you'd like you can add `--entrypoint=/bin/bash` and run 
+`/pwd/coreos_koji_tagger.py` directly. If you modify the json at the top
+of the file you can test out the import locally.
+
+
+# Running rabbitmq server locally:
+
+
+## Server
+
+The rough steps for setting up a server are: 
+
+- `sudo dnf install -y fedora-messaging rabbitmq-server`
+- `sudo systemctl start rabbitmq-server`
+
+Optional - to see a web browser view:
+
+- `sudo sed -i -e 's|@RABBITMQ_USER@|rabbitmq|' -e 's|@RABBITMQ_GROUP@|rabbitmq|' /usr/sbin/rabbitmq-plugins`
+    - https://bugzilla.redhat.com/show_bug.cgi?id=1755152
+- `sudo rabbitmq-plugins enable rabbitmq_management`
+- Navigate to `<IP_OF_HOST>:15672` in a web browser and log in with `guest`/`guest`. 
+- Navigate to `Queues` tab to view existing queues/messages.
+
+## Fedora Messaging consumer
+
+If you want to see the `request.ostree-import.finished` messages sent by the ostree-importer
+you can run the following command on the on the same system that is running the rabbitmq server.
+
+```
+fedora-messaging consume --callback=fedora_messaging.example:printer --routing-key org.fedoraproject.prod.coreos.build.request.ostree-import.finished
+```
+
+## Fedora Messaging sender
+
+If you'd like to send a `request.ostree-import` message to rabbitmq (i.e. letting the
+ostree-importer listen and react to the message) you can do something like this python file
+on the rabbitmq server:
+
+```
+cat <<'EOF' > publisher.py
+#!/usr/bin/python3
+from fedora_messaging import api, message
+topic = 'org.fedoraproject.prod.coreos.build.request.ostree-import'
+body = {
+    "build_id": "30.20190905.0",
+    "stream": "testing",
+    "basearch": "x86_64",
+    "commit": "s3://fcos-builds/prod/streams/testing/builds/30.20190905.0/x86_64/ostree-commit.tar",
```

Comment:
@jlebon any chance we could just make the ACL for this public and use an https url here instead so we don't have to worry about creating read-only credentials for AWS for this service?

Reply:
AFAIU, there should already be credentials available to use. See e.g. https://pagure.io/fedora-infrastructure/issue/8189#comment-600735 for a similar concern with robosignatory. Hmm, although that's probably not accessible to OpenShift apps.

If not, sure we can just use `https://`. The object is already public, but I guess I want to keep the door open for uploading bucket objects as private. But yeah... if we really wanted to do this, we could request credentials then.

Reply:
> AFAIU, there should already be credentials available to use. See e.g. https://pagure.io/fedora-infrastructure/issue/8189#comment-600735 for a similar concern with robosignatory. Hmm, although that's probably not accessible to OpenShift apps.

right. I think I would want a separate set of credentials for this app that were read-only.

> 
> If not, sure we can just use `https://`. The object is already public, but I guess I want to keep the door open for uploading bucket objects as private. But yeah... if we really wanted to do this, we could request credentials then.

It would be nice if we can do `https://` so I don't have to ask for new creds and also it will reduce the complexity of coreos-ostree-importer. Maybe we can do that soonish.